src/components/core/Markdown/Bold.tsx

```diff
@@ -0,0 +1,87 @@
+import IconContainer from './IconContainer';
+import { useCallback } from 'react';
+
+export default function Bold({
+  selectedRange,
+  textAreaRef,
+  setSelectedRange,
+  setUserDragged,
+}: {
+  selectedRange: number[];
+  textAreaRef: React.RefObject<HTMLTextAreaElement>;
+  setSelectedRange: (value: number[] | ((prevVar: number[]) => number[])) => void;
+  setUserDragged: (value: boolean) => void;
+}) {
+  const handleClick = useCallback(() => {
+    setUserDragged(false);
+
+    const textArea = textAreaRef.current;
+    if (textArea) {
+      const [start, end] = selectedRange;
+      const selectedText = textArea.value.substring(start, end);
+      const selectedTextWithSurrounding = textArea.value.substring(start - 2, end + 2);
+
+      const selectedTextIsBold = selectedText.startsWith('**') && selectedText.endsWith('**');
+      const selectedTextIsSurroundedBold =
+        selectedTextWithSurrounding.startsWith('**') && selectedTextWithSurrounding.endsWith('**');
+
+      // If the selected text includes any bold tags, remove them
+      if (selectedTextIsBold) {
+        const newText = textArea.value.replace(selectedText, selectedText.slice(2, -2));
+        textArea.value = newText;
+        setSelectedRange([start, end - 4]); // **Bold** ([0, 8]) -> Bold ([0, 4])
+        return;
+      }
+      if (selectedTextIsSurroundedBold) {
+        const newText = textArea.value.replace(
+          selectedTextWithSurrounding,
+          selectedTextWithSurrounding.slice(2, -2)
+        );
+        textArea.value = newText;
+        setSelectedRange([start - 2, end - 2]); // **Bold** ([2, 6]) -> Bold ([0, 4])
+        return;
+      }
+      if (selectedText.length > 0) {
+        // If the selected text is not already bold, add it
+        const newText =
+          textArea.value.substring(0, start) +
+          `**${selectedText}**` +
```

Comment:
since we're using template literals for `selectedText`, maybe we could use it for the whole value of `newText`?

or does it not fit in one line and look ugly 👀